mule-extensions-api/src/main/java/org/mule/runtime/extension/api/util/ExtensionModelUtils.java

```diff
@@ -268,6 +270,39 @@ private void resolve(ComponentModel model, Object owner) {
     return result.get();
   }
 
+  /**
+   * @param extensionModel the model which owns the {@code component}
+   * @param component      a component
+   * @return A {@link Set} with the {@link ConfigurationModel} that the can be used alongside with the {@code component}
+   */
+  public static Set<ConfigurationModel> getConfigurationForComponent(ExtensionModel extensionModel, ComponentModel component) {
+    Set<ConfigurationModel> result = new HashSet<>();
+    new ExtensionWalker() {
+
+      @Override
+      public void onOperation(HasOperationModels owner, OperationModel model) {
+        resolve(model, owner);
+      }
+
+      @Override
+      public void onSource(HasSourceModels owner, SourceModel model) {
+        resolve(model, owner);
+      }
+
+      private void resolve(ComponentModel model, Object owner) {
+        if (model == component && owner != extensionModel) {
+          result.add((ConfigurationModel) owner);
+        }
+      }
+    }.walk(extensionModel);
+
+    if (component instanceof ConnectableComponentModel && ((ConnectableComponentModel) component).requiresConnection()) {
+      extensionModel.getConfigurationModel(DEFAULT_CONFIG_NAME).ifPresent(result::add);
```

Comment:
this doesn't look right at all. For many connectors it will yield wrong information. What are you trying to cover here?

Reply:
i already migrated away from this code in the previous pr, this is just reintroducing it for backwards compatibility

Reply:
ref: https://github.com/mulesoft/mule-extensions-api/blob/1.8.2/mule-extensions-api/src/main/java/org/mule/runtime/extension/api/util/ExtensionModelUtils.java#L276